resources/lib/library.py

```diff
@@ -1500,9 +1501,9 @@ def explorer( self, history, location, label, thumbnail, itemType, isWidget = Fa
         widgetType = None
         addonType = None
 
-        dialogLabel = label[0].replace( "  >", "" )
+        dialogLabel = try_decode( label[0] ).replace( "  >", "" )
         if len( label ) != 1:
-            dialogLabel = label[0].replace( "  >", "" ) + " - " + label[ len( label ) - 1 ].replace( "  >", "" )
+            dialogLabel = try_decode( label[0] ).replace( "  >", "" ) + " - " + try_decode( label[ len( label ) - 1 ] ).replace( "  >", "" )
 
```

Comment:
pythonic way for "label[ len( label ) - 1 ]" is "label[:-1]" (you want to remove the last char, right?)